### PR TITLE
CFE-3625: Increase Filedescriptor Limit to a more contemporary Value

### PR DESCRIPTION
**Pull request by flynn1973**
Ticket: CFE-3625
Changelog: Title

### CODE REVIEW

**Comment by olehermanse** [comment]
@cf-bottom please test this in our Jenkins CI system :)

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/6470/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/6470/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-6470/

File: libpromises/pipes_unix.c [hunk 1]
```diff
@@ -39,7 +39,7 @@ static bool CfSetuid(uid_t uid, gid_t gid);
 static int cf_pwait(pid_t pid);
 
 static pid_t *CHILDREN = NULL; /* GLOBAL_X */
-static int MAX_FD = 128; /* GLOBAL_X */ /* Max number of simultaneous pipes */
+static int MAX_FD = 8192; /* GLOBAL_X */ /* Max number of simultaneous pipes */
```
Comment (vpodzime) [inline libpromises/pipes_unix.c:42]: Wow, big hammer approach. :) I'm not quite sure 8192 child processes (or even 4096) running at the same time is feasible. What about 2048?

**Review by vpodzime — COMMENTED** [review]
Looks good to me otherwise.

**Comment by vpodzime** [comment]
Just a note: I'm currently working on https://github.com/cfengine/libntech/pull/130 which should be a better long-term solution for this and many other similar problems. Having said that, we should definitely make this simple change first.

File: libpromises/pipes_unix.c (hunk 1 above)
Comment (flynn1973) [reply libpromises/pipes_unix.c:42]: :) yeah, maybe a little bit overstated, i like to live dangerously. 2048 should be enough.

Comment (nickanderson) [reply libpromises/pipes_unix.c:42]: I was going to stuffiest 640k

Comment (flynn1973) [reply libpromises/pipes_unix.c:42]: hmm...basically i can patch this locally, as i am constantly building cfengine rpms for aix.  https://github.com/cfengine/libntech/pull/130 looks very "promising" to me, and i dont want to fire up a "numbers" flamewar :-). this problem seems to be a isolated thing (only me) anyway. nobody is doing stupid stuff like me *g*

Comment (vpodzime) [reply libpromises/pipes_unix.c:42]: 128 is definitely a very low and totally artificial limit. 2048 should be fine for everybody. :wink:

Comment (olehermanse) [inline libpromises/pipes_unix.c:42]: As suggested by others, let's go to 2048 for now:

```suggestion
static int MAX_FD = 2048; /* GLOBAL_X */ /* Max number of simultaneous pipes */
```

**Review by olehermanse — CHANGES_REQUESTED** [review]

**Comment by vpodzime** [comment]
@flynn1973 could you please squash the commits? Thanks!

**Comment by flynn1973** [comment]
superseeded by https://github.com/cfengine/core/pull/4615

**Pull request closed without merging**